portality/migrate/subjects/remove_duplicate_subjects.py

```diff
@@ -0,0 +1,85 @@
+""" Scroll through articles, removing duplicate subjects. No evidence of same in journals """
+
+import esprit
+from portality import models
+from datetime import datetime
+from portality.core import app
+from portality.util import unicode_dict
+
+batch_size = 1000
+
+
+def rem_dup_sub(write_changes=False):
+
+    # Connection to the ES index
+    conn = esprit.raw.make_connection(None, 'localhost', 9200, 'doaj')
+
+    write_batch = []
+    updated_count = 0
+    same_count = 0
+
+    # Scroll though all articles in the index
+    for a in esprit.tasks.scroll(conn, 'article'):
+        try:
+            article_model = models.Article(_source=a)
+
+            # Remove the duplicates
+            subjects = unicode_dict(article_model.bibjson().subjects())
+            newsub = []
+
+            for sub in subjects:
+                if sub not in newsub:
+                    newsub.append(sub)
+
+            if newsub != subjects:
+                updated_count += 1
+                if write_changes:
+                    article_model.data['bibjson']['subject'] = newsub
+                    article_model.prep()
+                    write_batch.append(article_model.data)
+            else:
+                same_count += 1
+
+        except ValueError:
+            # Failed to create model (this shouldn't happen!)
```

Comment:
Is this one pushed? I just see except ValueError: continue

It's ok if you've decided not to do that too, but since you've noted "done" I'm just wondering